odoo/sql_db.py

```diff
@@ -741,7 +741,11 @@ def connection_info_for(db_or_uri):
     :param str db_or_uri: database name or postgres dsn
     :rtype: (str, dict)
     """
-    app_name = "odoo-%d" % os.getpid()
+    if 'ODOO_PGAPPNAME' in os.environ:
+        # Using manual string interpolation for security reason and trimming at default NAMEDATALEN=63
+        app_name = os.environ['ODOO_PGAPPNAME'].replace('{pid}', str(os.getpid()))[0:63]
```

Comment:
@xmo-odoo because of security issues in python string interpolations I heard of a long time ago. I guess those are probably fixed by now but I tough it was safer to do it manually. Now, if you confirm that it's ok use python's string interpolation with uncontrolled user input then I'm fine with this !

Reply:
If it's for the sake of a one-liner, what about this ?

```python
app_name = (os.environ.get('ODOO_PGAPPNAME') or 'odoo-{pid}').replace('{pid}', str(os.getpid()))[:63]
```

Reply:
> I mean currently you can crash odoo if you provide a --addons-path which doesn't exist so...

doesn't mean it should be a standard

btw, providing a wrong addon path makes the server exit gracefully with a meaningful error message, I don't consider this as a crash.

> sure?

mhh sorry I don't understand what you mean